### PR TITLE
Drop support for Swift 5.8

### PR DESCRIPTION
**Pull request by FranzBusch**
# Motivation

We only support the last three Swift released versions which are at this time 5.9, 5.10 and 6.

# Modification

This PR drops anything related to Swift 5.8.

# Result

Version support aligned.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,7 +86,8 @@ SwiftNIO            | Minimum Swift Version
 `2.43.0 ..< 2.51.0` | 5.5.2
 `2.51.0 ..< 2.60.0` | 5.6
 `2.60.0 ..< 2.65.0` | 5.7
-`2.65.0 ...`        | 5.8
+`2.65.0 ..< 2.75.0  | 5.8
+`2.75.0 ...`        | 5.9
```
Comment (glbrntt) [inline README.md:90]: You've got an off-by-one error here: 2.75.0 has shipped and supports 5.8. Dropping 5.8 is a minor so this should be 2.76.0.

File: Sources/NIOCore/NIOScheduledCallback.swift [hunk 2]
```diff
@@ -89,8 +89,7 @@ public struct NIOScheduledCallback: Sendable {
 }
 
 extension EventLoop {
-    // This could be package once we drop Swift 5.8.
-    public func _scheduleCallback(
+    package func _scheduleCallback(
```
Comment (glbrntt) [inline Sources/NIOCore/NIOScheduledCallback.swift:92]: Should we drop the `_` here?

File: README.md [hunk 3]
```diff
@@ -86,7 +86,8 @@ SwiftNIO            | Minimum Swift Version
 `2.43.0 ..< 2.51.0` | 5.5.2
 `2.51.0 ..< 2.60.0` | 5.6
 `2.60.0 ..< 2.65.0` | 5.7
-`2.65.0 ...`        | 5.8
+`2.65.0 ..< 2.75.0  | 5.8
```
Comment (Lukasa) [inline README.md:89]: This is still wrong, it's 
```suggestion
`2.65.0 ..< 2.76.0  | 5.8
```

**Comment by FranzBusch** [comment]
@glbrntt I kept the `_scheduleCallback` underscored since there is another `scheduleCallback` method on EL already that would conflict otherwise. That methods actually calls the underscored method.

**Review by rnro — APPROVED** [review]

**Review by Lukasa — APPROVED** [review]

**Pull request merged**